### PR TITLE
GEOMESA-1620 Refactor manual to organize and group by backend.

### PR DESCRIPTION
**Pull request by mdzimmerman**
* Organizes and groups content by backend (moved Accumulo specifics into
  "accumulo" directory, for example)
* Breaks down content into shorter pages.
* Organized other modules into "analytics" and "apis" directories.
* Created a separate directory with install instructions, also
  broken down by backend.
* Added a "Getting Started" page with links to tutorials.

Signed-off-by: Matthew Zimmerman <matt.zimmerman@ccri.com>

### CODE REVIEW

**Comment by mdzimmerman** [comment]
Built version here: http://www.geomesa.org/documentation/reorg2017/

File: docs/user/installation/hbase.rst [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+Due to licensing restrictions, dependencies for shape file support and raster
+ingest must be separately installed. Do this with the following commands:
+
+.. code-block:: bash
+
+    $ bin/install-jai.sh
+    $ bin/install-jline.sh
+
+Test the command that invokes the GeoMesa Tools:
+
```
Comment (aheyne) [inline docs/user/installation/hbase.rst]: Is something missing here?

File: docs/user/kafka/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+however, a ``geomesa-site.xml`` configuration file is available for
+setting system properties.
+
+This file can be found at ``conf/geomesa-site.xml`` of the Kafka
+distribution. The default settings for GeoMesa are stored in
+``conf/geomesa-site.xml.template``. Do not modify this file directly as it
+is never read; instead, copy the desired configurations into
+``geomesa-site.xml``.
+
+By default command line parameters will take precedence over this
```
Comment (aheyne) [inline docs/user/kafka/configuration.rst]: Should be comma after default (introductory phrase)

File: docs/user/kafka/configuration.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+configuration file. If you wish a configuration item to always take
+precedence, even over command line parameters, change the ``<final>``
+tag to true.
+
+By default configuration properties with empty values will not be
```
Comment (aheyne) [inline docs/user/kafka/configuration.rst]: same

File: docs/user/kafka/consumers.rst [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+Feature Source to load, at initialization time, all state changes that
+occur within the window. Any query for a time outside of the window will
+return no results even if features existed at that time.
+
+The read behind is the amount of time used to rebuild state. For
+example, if ``readBehind = 5s`` then for a query requesting state at
+``time = t`` all state changes that occurred between ``t - 5s`` and
+``t`` will be used to build the state at time ``t`` which will then be
+used to answer the query. Selecting an appropriate read behind requires
+an understanding of the producer. The expected uses case is a producer
```
Comment (aheyne) [inline docs/user/kafka/consumers.rst]: /uses/use/

File: docs/user/kafka/consumers.rst [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+During initialization of the Kafka Replay Feature Source all state
+changes from ``replayStart - readBehind`` to ``replayEnd`` will be read
+and cached. As the size of the replay window and read behind increases
+so does the amount of data that must be read and cashed. So, both the
+size of the window and the read behind should be kept as small as
+possible.
+
+After creating the ``ReplayConfig`` pass it, along with the
+``streamingSFT`` to the ``KafkaDataStoreHelper``:
```
Comment (aheyne) [inline docs/user/kafka/consumers.rst]: Awkward comma, maybe?: 
After creating the ``ReplayConfig``, pass it, along with the ``streamingSFT``, to the ``KafkaDataStoreHelper``:

File: docs/user/kafka/consumers.rst [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    SimpleFeatureType streamingSFT = consumerDs.getSchema(typeName);
+    SimpleFeatureType replaySFT = KafkaDataStoreHelper.createReplaySFT(streamingSFT, replayConfig);
+
+The ``streamingSFT`` passed to ``createReplaySFT`` must contain the
+hints added by ``KafkaDataStoreHelper.createStreamingSFT``. The easiest
+way to ensure this is to call ``consumerDs.getSchema(typeName)``. The
+``SimpleFeatureType`` returned by ``createReplaySFT`` will contain the
+hint added by ``createStreamingSFT`` as well as a a hint containing the
+``ReplayConfig``. Additionally the ``replaySFT`` will have a different
+name then then ``streamingSFT``. This is to differentiate *live* and
```
Comment (aheyne) [inline docs/user/kafka/consumers.rst]: /then then/than/

File: docs/user/cassandra/commandline_tools.rst [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+  Edit this argument if needed to match the name of your key space.
+- The ``catalog`` argument specifies the name of a Cassandra table that will be created in your name space
+  by the ``ingest`` command.
+  You can name the catalog table anything you want.
+  This table is a metadata table that contains a list of each GeoMesa data table
+  that you have created. Before you run the ``ingest`` command for the first time, this table does
+  not exist. When you run ``ingest`` for the first time, GeoMesa creates two tables: this "catalog" metadata
+  table, and the table that actually stores your data. If you run ``ingest`` again with a different dataset,
+  GeoMesa will append a row to the "catalog" table, and also create an additional table for the new data.
+- The ``name-space`` parameter can be anything you want. (It is used by GeoTools to provide a name space for your feature types.)
```
Comment (mdzimmerman) [inline docs/user/cassandra/commandline_tools.rst]: Agreed, but this is not a misspelling, the option passed to ``geomesa-cassandra`` actually *is* ``--name-space``. We could change the tool's options, but IMHO that should be a separate PR.

File: docs/user/cassandra/commandline_tools.rst [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+- The ``converter`` and ``spec`` parameters refer to a specific file located at ``conf/application.conf`` within the
+  ``GEOMESA_CASSANDRA_HOME`` directory. This file contains specifications for "converters" and "sfts" (SimpleFeatureTypes).
+  In this specific example it only contains one of each: a converter and a SimpleFeatureType which are both called
+  "example_csv". The converter specifies how a raw data file should be parsed. For example, the converter specifies
+  that the fourth column of the input file should be converted to a date using a specific date format. The
+  SimpleFeatureType part specifies how the parsed data should be used to create a SimpleFeatureType. For example, this
+  files specifies that the SimpleFeatureType's first attribute should be one called "name". GeoMesa uses these specifications
```
Comment (mdzimmerman) [inline docs/user/cassandra/commandline_tools.rst]: fixed

File: docs/user/cassandra/commandline_tools.rst [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+        --feature-name example_csv
+
+Configuring the Command Line Tools
+----------------------------------
+
+You can configure the command line tools using the
+``conf/geomesa-env.sh`` file in the ``GEOMESA_CASSANDRA_HOME`` directory.
+See the comments in that file for instructions.
+
+
```
Comment (mdzimmerman) [inline docs/user/cassandra/commandline_tools.rst]: fixed

File: docs/user/cassandra/contribute.rst [hunk 10]
```diff
@@ -0,0 +1,15 @@
+How you can contribute
+======================
+
+Here's a list of items that we will be adding to optimize the Cassandra DataStore:
```
Comment (mdzimmerman) [inline docs/user/cassandra/contribute.rst]: fixed

File: docs/user/cassandra/index.rst [hunk 11]
```diff
@@ -0,0 +1,17 @@
+Cassandra Data Store
+====================
+
+The GeoMesa Cassandra Data Store is found in the ``geomesa-cassandra``
```
Comment (mdzimmerman) [inline docs/user/cassandra/index.rst]: yep. fixed

File: docs/user/installation/cassandra.rst [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+After you've successfully installed and configured Cassandra, you can
+move on to installing the GeoMesa Cassandra distribution.
+Start by choosing or creating a directory where you want
+to put it. Then ``cd`` into that directory and type::
+
+    wget https://repo.locationtech.org/content/repositories/geomesa-releases/org/locationtech/geomesa/geomesa-cassandra-dist_2.11/$VERSION/geomesa-cassandra-dist_2.11-$VERSION-bin.tar.gz
+    tar xvf geomesa-cassandra-dist_2.11-$VERSION-bin.tar.gz
+
+(Make sure to replace ``$VERSION`` with the version of the distribution that you want to use.
```
Comment (mdzimmerman) [inline docs/user/installation/cassandra.rst]: fixed

File: docs/user/installation/hbase.rst (hunk 1 above)
Comment (mdzimmerman) [inline docs/user/installation/hbase.rst]: ack, cut and paste fail. fixed

Comment (mdzimmerman) [inline docs/user/installation/hbase.rst]: In fact this whole section needs rewritten. Good catch!

File: docs/user/kafka/configuration.rst (hunk 2 above)
Comment (mdzimmerman) [inline docs/user/kafka/configuration.rst]: fixed

File: docs/user/kafka/configuration.rst (hunk 3 above)
Comment (mdzimmerman) [inline docs/user/kafka/configuration.rst]: fixed

File: docs/user/kafka/consumers.rst (hunk 4 above)
Comment (mdzimmerman) [inline docs/user/kafka/consumers.rst]: fixed

File: docs/user/kafka/consumers.rst (hunk 5 above)
Comment (mdzimmerman) [inline docs/user/kafka/consumers.rst]: yeah, I think that was the intention

File: docs/user/kafka/consumers.rst (hunk 6 above)
Comment (mdzimmerman) [inline docs/user/kafka/consumers.rst]: fixed

File: docs/user/installation/hbase.rst (hunk 1 above)
Comment (elahrvivaz) [inline docs/user/installation/hbase.rst]: oops, yeah there are lots of accumulo references

File: docs/common.py [hunk 13]
```diff
@@ -80,6 +84,12 @@
 
 .. |release_tarball_kafka10| replace:: %(url_locationtech_release)s/geomesa-kafka-10-dist_2.11/%(release)s/geomesa-kafka-10-dist_2.11-%(release)s-bin.tar.gz
 
+.. |release_tarball_hbase| replace:: %(url_locationtech_release)s/geomesa-hbase-dist_2.11/%(release)s/geomesa-hbase-dist_2.11-%(release)s-bin.tar.gz
+
+.. |release_tarball_bigtable| replace:: %(url_locationtech_release)s/geomesa-bigtable-dist_2.11/%(release)s/geomesa-bigtable-dist_2.11-%(release)s-bin.tar.gz
```
Comment (elahrvivaz) [inline docs/common.py]: this might not be used anywhere so it might not matter - but we don't release the bigtable dist on locationtech

File: docs/user/cassandra/commandline_tools.rst (hunk 7 above)
Comment (jnh5y) [inline docs/user/cassandra/commandline_tools.rst]: Does C* hyphenate it for some reason?

File: docs/user/cassandra/commandline_tools.rst [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+  You can name the catalog table anything you want.
+  This table is a metadata table that contains a list of each GeoMesa data table
+  that you have created. Before you run the ``ingest`` command for the first time, this table does
+  not exist. When you run ``ingest`` for the first time, GeoMesa creates two tables: this "catalog" metadata
+  table, and the table that actually stores your data. If you run ``ingest`` again with a different dataset,
+  GeoMesa will append a row to the "catalog" table, and also create an additional table for the new data.
+- The ``name-space`` parameter can be anything you want. (It is used by GeoTools to provide a name space for your feature types.)
+- The ``converter`` and ``spec`` parameters refer to a specific file located at ``conf/application.conf`` within the
+  ``GEOMESA_CASSANDRA_HOME`` directory. This file contains specifications for "converters" and "sfts" (SimpleFeatureTypes).
+  In this specific example it only contains one of each: a converter and a SimpleFeatureType which are both called
```
Comment (jnh5y) [inline docs/user/cassandra/commandline_tools.rst]: Should this discussion of the converters link to the Converter's documentation?

File: docs/user/bigtable/index.rst [hunk 15]
```diff
@@ -0,0 +1,16 @@
+Bigtable Data Store
+===================
+
+Experimental support for GeoMesa implemented on `Google Cloud Bigtable <https://cloud.google.com/bigtable>`__.
```
Comment (jnh5y) [inline docs/user/bigtable/index.rst]: Drop 'experimental';).

File: docs/user/cassandra/commandline_tools.rst [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+  "example_csv". The converter specifies how a raw data file should be parsed. For example, the converter specifies
+  that the fourth column of the input file should be converted to a date using a specific date format. The
+  SimpleFeatureType part specifies how the parsed data should be used to create a SimpleFeatureType. For example, this
+  file specifies that the SimpleFeatureType's first attribute should be one called "name". GeoMesa uses these specifications
+  to ingest the data from the external data file into the Cassandra database.
+
+  GeoMesa automatically finds the ``conf/application.conf`` file based on its name and location. You can add additional
+  converters and SFT specifications to it as needed for ingesting other datasets. In addition, you can also
+  specify converters and SFT specifications by adding directories to the
+  ``conf/sfts`` directory. For more details see :ref:`installing_sft_and_converter_definitions`.
```
Comment (jnh5y) [inline docs/user/cassandra/commandline_tools.rst]: Ok, so it does link out to it.  I wonder if the discussion should be tightened up or dropped?

File: docs/user/cassandra/commandline_tools.rst [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+- The last argument is the location of the data file that we want to ingest.
+
+If needed, edit the parameter arguments in the bash script. Then, ``cd`` into the ``GEOMESA_CASSANDRA_HOME``
+directory, and run the script by typing::
+
+  source examples/ingest/csv/ingest_csv.sh
```
Comment (jnh5y) [inline docs/user/cassandra/commandline_tools.rst]: Are we 'source'ing this rather than just running it for some reason?

File: docs/user/cassandra/commandline_tools.rst (hunk 7 above)
Comment (elahrvivaz) [inline docs/user/cassandra/commandline_tools.rst]: this should shake out with the updated C* impl. Right now it's a bit non-standard (compared to our other back-ends)

File: docs/user/installation/hbase.rst [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+        export             Export features from a GeoMesa data store
+        export-bin         Export features from a GeoMesa data store in a binary format
+        gen-avro-schema    Generate an Avro schema from a SimpleFeatureType
+        get-sft-config     Get the SimpleFeatureType of a feature
+        get-type-names     List GeoMesa feature types for a given catalog
+        help               Show help
+        ingest             Ingest/convert various file formats into GeoMesa
+        keywords           Add/Remove/List keywords on an existing schema
+        remove-schema      Remove a schema and associated features from a GeoMesa catalog
+        stats-analyze      Analyze statistics on a GeoMesa feature type
```
Comment (jnh5y) [inline docs/user/installation/hbase.rst]: HBase + stats?!  @elahrvivaz

File: docs/user/kafka/configuration.rst [hunk 19]
```diff
@@ -0,0 +1,20 @@
+Configuration
+=============
+
+Use of the Kafka data store does not require server-side configuration,
```
Comment (jnh5y) [inline docs/user/kafka/configuration.rst]: What Kafka configuration is in the geomesa-site.xml?  Is it worth mentioning those options to motivate checking it out for the KDS user?

Comment (elahrvivaz) [inline docs/user/kafka/configuration.rst]: i'm working on a better configuration section - i'll cover kds there too

File: docs/common.py (hunk 13 above)
Comment (mdzimmerman) [inline docs/common.py]: Fixed.

File: docs/user/bigtable/index.rst (hunk 15 above)
Comment (mdzimmerman) [inline docs/user/bigtable/index.rst]: Fixed.

File: docs/user/cassandra/commandline_tools.rst (hunk 17 above)
Comment (mdzimmerman) [inline docs/user/cassandra/commandline_tools.rst]: I have not run this myself, but all it's doing is executing ``geomesa-cassandra``, so I think that just running it would work.

File: docs/user/cassandra/commandline_tools.rst (hunk 14 above)
Comment (mdzimmerman) [inline docs/user/cassandra/commandline_tools.rst]: Sure.

File: docs/user/cassandra/commandline_tools.rst (hunk 16 above)
Comment (mdzimmerman) [inline docs/user/cassandra/commandline_tools.rst]: I just added a couple of links to the converter documentation for now.

File: docs/user/installation/hbase.rst (hunk 18 above)
Comment (mdzimmerman) [inline docs/user/installation/hbase.rst]: Well, that's what you get when you run ``geomesa-hbase``! Didn't actually check each commands to see if they would do anything.
